sdk/metric/pipeline.go

```diff
@@ -175,7 +185,72 @@ func newPipelineRegistries(views map[Reader][]view.View) *pipelineRegistry {
 	return &pipelineRegistry{
 		views:     views,
 		pipelines: pipelines,
+
+		cacheInt:   map[instrumentID]pipelineCacheResult[int64]{},
+		cacheFloat: map[instrumentID]pipelineCacheResult[float64]{},
+	}
+}
+
+func (reg *pipelineRegistry) createInt64Aggregators(inst view.Instrument, instUnit unit.Unit) ([]internal.Aggregator[int64], error) {
+	reg.cacheLock.Lock()
+	defer reg.cacheLock.Unlock()
+	key := instrumentID{
+		scope:       inst.Scope,
+		name:        inst.Name,
+		description: inst.Description,
+	}
+
+	if result, ok := reg.cacheInt[key]; ok {
+		if inst.Kind != result.instrumentKind {
+			return nil, fmt.Errorf("instruments collide")
+		}
+		return result.aggregators, result.err
+	}
+
+	aggs, err := createAggregators[int64](reg, inst, instUnit)
+	reg.cacheInt[key] = pipelineCacheResult[int64]{
+		aggregators:    aggs,
+		err:            err,
+		instrumentKind: inst.Kind,
+	}
+	missedCacheInstrumentKind := view.InstrumentKind(254)
```

Comment:
This magic number should probably be a named constant for clarity. Indeed, I think `missedCacheInstrumentKind` itself could be a package-level constant.